tests/file.txt.bz2.test

```diff
@@ -0,0 +1 @@
+file.txt.bz2
```

Comment:
This file is not binary (according to github). So you are not actually testing the opening of binary files.

Reply:
I think this is just GitHub not displaying the symlink correctly.

But it would actually be an idea not to add any files at all, but to instead just create a temporary copy the existing file as part of the test. Then the repo itself isn’t cluttered with the symlinks.

Reply:
Ah thanks! But why create temporary files? These test files can be opened as is, right? As far as I can see the tests do nothing special (with regards to opening).

Reply:
To test whether the file format is recognized even though the file name extension is missing, we need (among others) a gzip file whose name does not end in `.gz`. We can get that the way it’s done currently: Add an extra file or symbolic link that has the appropriate name to the tests/ folder permanently. What I’m suggesting is that we could instead create the file each time the test is run from one of the existing files in the tests/ folder.

Or are we talking about different things?

Reply:
> To test whether the file format is recognized even though the file name extension is missing, we need (among others) a gzip file whose name does not end in .gz. 

Ah I see. Then adding links makes sense.

But since we can detect the file format by using bytes magic, does it make sense to keep the `.gz` recognition stuff? That is just extra maintenance load. It is been made redundant by the much more universal bytes magic.

Reply:
Yes, I think this could be changed to use content detection first if possible. However, we need `_detect_format_from_extension` anyway for when we’re writing to a file, and as a fallback for those cases where the file is actually not a regular file. For example, it could be a FIFO that has the `.gz` extension (I have done this, so it’s not entirely unrealistic ...). Let’s do that separately from this PR, though.